### DISCUSSION TITLE
Propose a script to create

### DISCUSSION THREAD

**Discussion by aleff-github** [aleff-github/my-flipper-shits#1] [Ideas]
## Description
Thank you to those who decide to participate.
I am currently working on creating a wide range of payloads that can be used in the BADUSB mode, whatever ideas you have in mind please write them down. 

## Legend
- 🟢 Totally - You must do nothing
- 🟡 Partial - Just something
- 🔴 Manual effort request

**Comment by plyght** [comment]
Hello,

I wondered if a MacOS badUSB is in the works, as I saw you don't have a Mac.

If you'd be interested, I have a Mac at my disposal to help you make one.

If not, no worries, and thanks for your time!

Best, plyght

**Reply by aleff-github** [reply]
> Hello,

Hi thanks for writing.

> I wondered if a MacOS badUSB is in the works, as I saw you don't have a Mac.

Creating a payload for a certain device necessarily requires owning the device because testing is always necessary.

In the past I tried to buy a second-hand Mac on a refurbished computer site, but unfortunately I experienced firsthand the programmed obsolescence of Mac computers in that, despite the splendid hardware, it was virtually unusable since the applications were almost all updated to a version of macOS that was not available on the computer I had taken because it was too old and almost all of the applications were not backwards compatible (I could only use Safari and Notes). This total limitation forced me to have to return the item as it was unusable.

>If you'd be interested, I have a Mac at my disposal to help you make one.
If not, no worries, and thanks for your time!

I have told you this long story to tell you that I could never buy a new Mac computer (given the cost and since I would never use it as my main operating system) just to do some testing or to create payloads in DuckyScript and, therefore, I provided a [donation system to support me by allowing me to buy one](https://github.com/sponsors/aleff-github/sponsorships?sponsor=aleff-github&tier_id=279522&preview=false).

In this specific case, since you are proposing a physical donation and not a financial one, I say to you that if the computer you have has not already undergone the fateful planned obsolescence and supports a version of macOS that allows the use of software without major problems then it may be fine, otherwise I thank you for the thought but it may not be usable for the intended purposes.

> Best, plyght

Thanks to you, Aleff ^^

**Reply by plyght** [reply]
hello,
thanks for the quick response!

I definitely don't have the money for that either lmao; I have tons of relatively old MacBooks that I don't use.
I would probably prefer if I could keep them and test them at my home, but if you'd really like them, I might consider shipping them.

LMK,
plyght

**Reply by aleff-github** [reply]
In the previous message, I was so precise because I wanted to create a valid response even for those who may have the same doubt in the future.

At this point, I believe the best thing for you is to keep your computer and try yourself to get involved by creating some payloads.

I want to tell you two things:
- The first is that often in this world, you won't find exactly what you're looking for, and almost always, it's necessary to make adjustments that fit your target (sometimes it's as simple as adapting the language). So, getting into coding is always the best solution to learn and get what you want.
- I sincerely thank you for considering the possibility of giving me your Mac because I appreciate the thought alone. Unfortunately, the costs of some devices don't allow for hacking (hacking in the broadest sense of the term, for more details, see the book "[HACKERS - The Heroes of the Computer Revolution](https://en.m.wikipedia.org/wiki/Hackers:_Heroes_of_the_Computer_Revolution)"), so I'm much happier knowing that from now on, you'll be trying your hand at hacking by creating your custom scripts instead of receiving a donation or a gifted computer. Hacking is priceless, and spreading it is worth more than anything else.

Happy Hacking, Aleff.

**Comment by the420man77** [comment]
Can you create a payload that will give full remote access to an iPhone for someone to find out if their spouse is involved in infidelity or not? It could be the potential gift that someone may need in order to help make the decision to potentially change their life for the better?! It could answer many questions that someone may be in total suffrage by and potentially give them the answers they need to continue on positively with their existence?!

**Reply by aleff-github** [reply]
I am completely opposed to creating a script or any kind of software that could violate people's privacy or allow unauthorized access to their devices. Such actions are ethically wrong and may be illegal. Privacy and ethics are fundamental principles that should guide our digital actions. I cannot assist you with this type of request.

Furthermore, if anyone reading this message believes they are under cyberattacks or that criminals may have gained remote access to their devices, it is crucial to report this to the relevant authorities in your respective countries, such as the police. Your safety and security should always be a top priority.

For more info about it read:
- [Mobile devices](https://www.eff.org/it/taxonomy/term/11400)
- [Protecting Your Privacy if Your Phone is Taken Away](https://www.eff.org/deeplinks/2020/06/protecting-your-privacy-if-your-phone-taken-away)
- [Privacy about EFF](https://www.eff.org/issues/privacy)

**Reply by OMEGAALPHA1791** [reply]
Happy theres  some of us left in the world of hacking  that  have code (moral ones) not just write code thanks for sharing. Yet im not opposed to hacking IOS in any form just  against using it in the way said above. 
 thanks to all that make this whole flipper and codes possible       yours true fully Brett or AALPHAOOMEGA

**Comment by Fapp3r3Lit3** [comment]
write up a full encrypted one with as far as take out the microphone and tracking. Create custom commands. I have one to open my phone by touch accessability to open it up with saying "iPhone up" < that word can be anything > 

And I have one that gos straight to master reset.  It adding like a "lock my messages" with whatever word they choose. Or. Delete all and if possible uninstall. A certain app. Tele... signal ahem. lol. It would be great if the fuzzer app just totally broke up a signsl if the blues lights pulled up behind you so they could t radio anything in.

**Reply by aleff-github** [reply]
> write up a full encrypted one

What do you mean? The script cannot be encrypted since it has to be interpreted by the device. It can be encrypted a program to have to run but still you would need another application to decrypt it and that could be complex because if the application is not already there you would have to download and install it, unless you are dealing with computers that generally have something reusable

> with as far as take out the microphone and tracking.

What do you mean by removing the microphone and tracking? Turn off the microphone, disable it for some specific applications? By tracking do you mean third-party app profiling? Geolocation? Blocking the GPS functionality?

> Create custom commands. I have one to open my phone by touch accessability to open it up with saying "iPhone up" < that word can be anything >

This already is clearer in that you could refer to the Quick Commands app that almost all iPhones have now, were you referring to that?

> And I have one that gos straight to master reset.

With "Master Reset" I can think of some kind of formatting, were you referring to that?

> It adding like a "lock my messages" with whatever word they choose.

Again it is not clear what you are referring to here, blocking messages would imply blocking a certain signal or protocol for data flow... What does "blocking messages" mean, not being able to delete them? Also what do you mean by "messages"? SMS? Whatsapp messages? Telegram?

> Or. Delete all and if possible uninstall. A certain app. Tele... signal ahem

By "Delete All" do you mean delete the data of an application or delete the application itself? Empty the temporary files? Also you refer only to messaging applications, do you mean delete individual chats by any chance?

> It would be great if the fuzzer app just totally broke up a signsl

If the Fuzzer application breaks the signal? What does that mean?

> if the blues lights pulled up behind you so they could t radio anything in.

I did not understand what blue lights you are referring to or even what kind of radio connection you are referring to.

---

I read your request to create scripts, but I have to be honest: I didn't understand anything. Your text is completely disconnected and it seems that your only goal is to do damage or vandalize devices.

This portal is for proposing useful and sensible scripts, not for saying random words. I make myself available to create scripts free of charge upon request, but only if the ideas are well formulated and do not have the sole purpose of doing damage.

I spent time on your request only out of respect, but it did not deserve any at all. I invite you to rephrase your question clearly and precisely, specifying the purpose of the script and its functionality. Only then will I be able to help you.

**Reply by OMEGAALPHA1791** [reply]
I believed that he was talking about the police I think hes a bit paranoiod my friends

**Reply by teleds34** [reply]
ok i just ( like 3 months ) got into coding and more accurate hacking and i recently learnt about DuckyScript and badusbs and i was wondering if its possible to take all info accounts and passwords and such but i was confused with how to conseal  it better or like using a card reader ish device. Hold on let me rephrase my question ; can you create a DuckyScript that steals passwords and accounts but that camofloges it using something. again im new and not really sure of the capabilities of coding 

Sincerly,Teleds34 ( or whatever it is) 

thank you
